src/sagas/deposit.js

```diff
@@ -301,6 +304,50 @@ export function* getBitcoinAddress() {
   yield* autoSubmitDepositProof()
 }
 
+function createConfirmationChannel(deposit) {
+  return eventChannel((emit) => {
+    const listener = ({
+      transactionID,
+      confirmations,
+      requiredConfirmations,
+    }) => {
+      emit({ transactionID, confirmations, requiredConfirmations })
+
+      // Close channel once we have all the required confirmations
+      if (confirmations === requiredConfirmations) {
+        emit(END)
+      }
+    }
+
+    deposit.onReceivedFundingConfirmation(listener)
+
+    // no-op
+    // eventChannel subscriber must return an unsubscribe, but tbtc.js does not
+    // currently provide one
+    return () => {}
+  })
+}
+
+function* watchForFundingConfirmations(deposit) {
+  yield delay(500)
```

Comment:
What are these delays for? May be worth flagging with a comment as well.

Reply:
The delays are an attempt at debouncing. Actually I think this first delay doesn't really make sense, so I removed it as part of #355. It's an old way of doing it and maybe unnecessary? Here's what the [docs](https://redux-saga.js.org/docs/recipes/) say.